src/main/java/edu/hawaii/its/api/type/GroupingsAnnouncements.java

```diff
@@ -0,0 +1,61 @@
+package edu.hawaii.its.api.type;
+
+import edu.hawaii.its.api.util.JsonUtil;
+import edu.hawaii.its.api.wrapper.AttributesResult;
+
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class GroupingsAnnouncements {
+    private String resultCode;
+    private List<GroupingsAnnouncement> announcements = new ArrayList<>();
+
+    public GroupingsAnnouncements(List<GroupingsAnnouncement> announcements) {
+        if (announcements != null) {
+            for (GroupingsAnnouncement a : announcements) {
+                this.announcements.add(a);
+            }
+            setResultCode("SUCCESS");
+        }
+    }
+
+    public GroupingsAnnouncements() {
+        findAnnouncements(new ArrayList<>()); //need to have, cannot change to set?
+        setResultCode("FAILURE");
+    }
```

Comment:
On line 26, where you placed the "need to have" comment, it can be fixed by removing `= new ArrayList<>()` from the private announcements variable at the top and setting `announcements = new ArrayList<>()` at the start of your two constructors.

Reply:
I just got rid of the call to `findAnnouncements` in the empty param constructor, thanks!